common/include/pcl/experimental/executor/trait/index.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+namespace executor {
+
+/**
+ * \brief A given Executor can have a custom index for bulk execute.
+ *
+ * \details: By default if not explicitly specified by the executor the shape is std::size_t.
+ * The index represents the index of execution unit which is currently running.
+ *
+ * executor_index is an Executor type trait which provides the index type
+ * defined by the Executor.
+ *
+ * Any executor can define a custom index by defining the alias index_type for
+ * the custom index.
+ *
+ * Part of proposal P0443R13
+ */
+template <typename Executor, typename = void>
+struct executor_index {
+  using type = std::size_t;
```

Comment:
shouldn't this be pcl::index_t?

Reply:
Switched to uindex_t for all executors even for OMP since the index of a thread can't be negative.
I haven't changed the default behavior since the proposal specifies `std::size_t`.

Reply:
As suggested, besides OMP others are `index_t` and OMP being `int` until `sindex_t` is introduced.